lib/HALInterfaces/IIMU.h

```diff
@@ -0,0 +1,171 @@
+#define IIMU_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include <stdint.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/** Struct of the IMU data in x, y and z direction. */
+typedef struct _IMUData
+{
+    int16_t valueX; /* Raw sensor value in x direction (unitless) */
+    int16_t valueY; /* Raw sensor value in y direction (unitless) */
+    int16_t valueZ; /* Raw sensor value in z direction (unitless) */
+} __attribute__((packed)) IMUData;
+
+/** The abstract IMU interface.
+ * IMU stands for Inertial Measurement Unit.
+ */
+class IIMU
+{
+public:
+    /**
+     * Destroys the interface.
+     */
+    virtual ~IIMU()
+    {
+    }
+
+    /**
+     * Initializes the inertial sensors and detects their type.
+     *
+     * @return True if the sensor type was detected succesfully; false otherwise.
+     */
+    virtual bool init() = 0;
+
+    /**
+     * Enables all of the inertial sensors with a default configuration.
+     */
+    virtual void enableDefault() = 0;
+
+    /**
+     * 	Configures the sensors with settings optimized for turn sensing.
+     */
+    virtual void configureForTurnSensing() = 0;
+
+    /**
+     * Takes a reading from the accelerometer and makes the measurements available in a.
+     */
+    virtual void readAccelerometer() = 0;
+
+    /**
+     * Takes a reading from the gyro and makes the measurements available in g.
+     */
+    virtual void readGyro() = 0;
+
+    /**
+     * Takes a reading from the magnetometer and makes the measurements available in m.
+     */
+    virtual void readMagnetometer() = 0;
+
+    /**
+     * Indicates whether the accelerometer has new measurement data ready.
+     *
+     * @return True if there is new accelerometer data available; false otherwise.
+     */
+    virtual bool accelerometerDataReady() = 0;
+
+    /**
+     * Indicates whether the gyro  has new measurement data ready.
+     *
+     * @return True if there is new gyro data available; false otherwise.
+     */
+    virtual bool gyroDataReady() = 0;
+
+    /**
+     * Indicates whether the magnetometer  has new measurement data ready.
+     *
+     * @return True if there is new magnetometer data available; false otherwise.
+     */
+    virtual bool magnetometerDataReady() = 0;
+
+    /**
+     * Get last raw Accelerometer values as a IMUData struct containing values in x, y and z.
```

Comment:
Which unit do I get for the acceleration? Isn't it in g? Because you keep IMUData unitless to be able to use the same struct (not best practice), you should mention the unit here.

Reply:
Added an explanation why the value is unitless.